app/scripts/models/mixins/resume-token.js

```diff
@@ -44,16 +46,30 @@ define(function (require, exports, module) {
     },
 
     /**
-     * Sets model properties from a ResumeToken.
+     * Sets model properties from a ResumeToken, unless said token
+     * is invalid according to `this.resumeTokenSchema`.
      *
      * @method populateFromResumeToken
      * @param {ResumeToken} resumeToken
      */
     populateFromResumeToken: function (resumeToken) {
       if (this.resumeTokenFields) {
-        this.set(resumeToken.pick(this.resumeTokenFields));
+        var pickedResumeToken = resumeToken.pick(this.resumeTokenFields);
+
+        if (isResumeTokenValid(pickedResumeToken, this.resumeTokenSchema)) {
+          this.set(pickedResumeToken);
+        }
       }
     }
   };
+
+  function isResumeTokenValid (resumeToken, schema) {
+    if (schema) {
+      var result = vat.validate(resumeToken, schema);
```

Comment:
> Right now, I silently fail to populate the model. Wasn't sure what else to do, got any suggestions?

That seems like reasonable behavior, if the flowId cannot be fetched from the ResumeToken, one will be generated locally as a fallback? 

Local generation would handle 2 cases:
* ResumeToken or flowId withing ResumeToken is corrupt.
* `flowId` is not part of the ResumeToken because the token was generated pre-flowId.

Like I said above, logging the validation error would help us understand how frequently resume tokens are corrupt in the wild.